src/app/view/perspectives/table/body/rows/cell-group/data-cell/table-data-cell.component.ts

```diff
@@ -87,64 +93,125 @@ export class TableDataCellComponent implements OnInit, OnChanges, OnDestroy {
   @Input()
   public allowedPermissions: AllowedPermissions;
 
+  @Output()
+  public affect = new EventEmitter();
+
   @ViewChild(TableDataCellMenuComponent)
   public menuComponent: TableDataCellMenuComponent;
 
   @ViewChild(DocumentHintsComponent)
   public suggestions: DocumentHintsComponent;
 
-  @ViewChild(TableEditableCellDirective)
-  public editableCell: TableEditableCellDirective;
+  @HostBinding('class.affected')
+  public affected: boolean;
+
+  @HostBinding('class.edited')
+  public edited: boolean;
+
+  @HostBinding('style.width.px')
+  public columnWidth: number;
+
+  @HostBinding('class.table-border-right')
+  public tableBorderRight = true;
 
-  public affected$: Observable<boolean>;
+  @HostBinding('class.table-border-bottom')
+  public tableBorderBottom = true;
+
+  public editing$ = new BehaviorSubject(false);
   public suggesting$ = new BehaviorSubject(false);
 
-  public editedValue: string;
+  public constraint$: Observable<Constraint>;
+
+  public editedValue: any;
+  public hiddenInputValue$ = new BehaviorSubject<any>('');
 
   private selectedSubscriptions = new Subscription();
+  private subscriptions = new Subscription();
 
   private savingDisabled: boolean;
 
   public constructor(
     private actions$: Actions,
+    private changeDetector: ChangeDetectorRef,
```

Comment:
Is this even used?

Reply:
It was used before when I tried to run change detection here. But then I realized we need to run it in the parent component to refresh `HostBinding` properties in the view.